File: problems/problems_3648/Solution.java
package problems.problems_3648;

import com.alibaba.fastjson.JSON;
import java.util.*;
import qubhjava.BaseSolution;


public class Solution extends BaseSolution {
    public int minSensors(int n, int m, int k) {
        k = 2 * k + 1;
        return ((m - 1) / k + 1) * ((n - 1) / k + 1);
    }

    @Override
    public Object solve(String[] inputJsonValues) {
        int n = Integer.parseInt(inputJsonValues[0]);
		int m = Integer.parseInt(inputJsonValues[1]);
		int k = Integer.parseInt(inputJsonValues[2]);
        return JSON.toJSON(minSensors(n, m, k));
    }
}

File: problems/problems_3648/solution.go
package problem3648

import (
	"encoding/json"
	"log"
	"strings"
)

func minSensors(n int, m int, k int) int {
	k = 2*k + 1
	return ((m-1)/k + 1) * ((n-1)/k + 1)
}

func Solve(inputJsonValues string) any {
	inputValues := strings.Split(inputJsonValues, "\n")
	var n int
	var m int
	var k int

	if err := json.Unmarshal([]byte(inputValues[0]), &n); err != nil {
		log.Fatal(err)
	}
	if err := json.Unmarshal([]byte(inputValues[1]), &m); err != nil {
		log.Fatal(err)
	}
	if err := json.Unmarshal([]byte(inputValues[2]), &k); err != nil {
		log.Fatal(err)
	}

	return minSensors(n, m, k)
}
